Port meson projects away from buildRustPackage

jtojnar opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
﻿The new setup hooks compose better with Meson setup hooks.

Fixes: https://github.com/NixOS/nixpkgs/issues/114850

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

jtojnar committed:
fractal: port away from buildRustPackage

The new setup hooks compose better with Meson setup hooks.

Also correct license.

jtojnar committed:
contrast: port away from buildRustPackage

The new setup hooks compose better with Meson setup hooks.

Needed to add libsass – not sure how sass-sys crate built without it before.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 116324` at c8728f39 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/878a9da87c17341a43a1f986fda65ae4">fractal</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/networking/instant-messengers/fractal/default.nix:91:5:
```
   |
91 |     license = licenses.gpl3;
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/applications/networking/instant-messengers/fractal/default.nix:4:3:
```
  |
4 | , fetchpatch
  |   ^
```
</li>
  </ul>
</details>

jtojnar committed:
shortwave: port away from buildRustPackage

The new setup hooks compose better with Meson setup hooks.

Had to add Git since mkDerivation does not bring it in any more.

jtojnar committed:
gnome-podcasts: port away from buildRustPackage

The new setup hooks compose better with Meson setup hooks.

Also correct license.

SuperSandro2000 commented on pkgs/applications/accessibility/contrast/default.nix:
```diff
@@ -39,6 +44,9 @@ rustPlatform.buildRustPackage rec {
     ninja
     pkg-config
     python3
+    rustPlatform.rust.cargo
+    rustPlatform.cargoSetupHook
+    rustPlatform.rust.rustc
     wrapGAppsHook
     glib # for glib-compile-resources
   ];
```
```suggestion
    wrapGAppsHook
    glib # for glib-compile-resources
  ] ++ (with rustPlatform; [
    rust.cargo
    cargoSetupHook
    rust.rustc
  ];
```

This could be done for all 4 packages.

jtojnar replied:
For just three packages it just looks too clunky IMO so I tend to avoid `with` in my packages unless there are much more dependencies.

worldofpeace reviewed (approved):
All LGTM
